google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/HeaderInterceptor.java

```diff
@@ -174,6 +175,10 @@ private void processHeader(
         }
         if (span != null) {
           span.setAttribute("gfe_latency", String.valueOf(gfeLatency));
+          String reqId = XGoogSpannerRequestId.getRequestIdFromMetadata(metadata);
```

Comment:
@odeke-em Have you tested these changes, I dont think it will work, since request_id is a request header and not response header. 

I have added the similar changes in Exemplar work https://github.com/googleapis/java-spanner/pull/3997

Reply:
We encountered an oddity that the Java Spanner client tests for OpenTelemetry only work for certain versions of Java as @olavloite mentioned in our group, so even testing anything requires y'all to help me run the bots /cc @tharoldD to help out with the updates.